simplelru/expirable_lru.go

```diff
@@ -0,0 +1,276 @@
+package simplelru
+
+import (
+	"container/list"
+	"sync"
+	"time"
+)
+
+// ExpirableLRU implements a thread safe LRU with expirable entries.
+type ExpirableLRU struct {
+	size       int
+	purgeEvery time.Duration
+	ttl        time.Duration
+	done       chan struct{}
+	onEvicted  EvictCallback
+
+	sync.Mutex
+	items     map[interface{}]*list.Element
+	evictList *list.List
+}
+
+// noEvictionTTL - very long ttl to prevent eviction
+const noEvictionTTL = time.Hour * 24 * 365 * 10
```

Comment:
If a TTL of 0 is defined as turning TTL off, why do you need this at all? Couldn't the check below simply be `if res.ttl != 0 && ....`

Reply:
Current expiration logic is very straightforward: `ent.Value.(*expirableEntry).expiresAt = now.Add(c.ttl)`

Considering zero a special case after the cache creation would complicate the code I believe. I would rather have this extra complexity on cache instance setup rather than inside it's logic. I haven't thought of any more elegant alternative, but it definetly doesn't mean it's not possible to find it.

Reply:
@jefferai  would you be so kind as to look into discussions in this PR?

Reply:
@jefferai gentle ping